google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -259,11 +259,17 @@ public void close() {
             // Ignore as we are closing the connection.
           }
         }
-        statementExecutor.shutdownNow();
-        spannerPool.removeConnection(options, this);
+        // Try to wait for the current statement to finish (if any) before we actually close the
+        // connection.
+        statementExecutor.shutdown();
```

Comment:
Good point. Although I'm not sure it actually cause any problems, it would create an unpredictable situation, so it's better to set `closed = true` before doing anything else.